avidemux_core/ADM_coreImage/src/ADM_image.cpp

```diff
@@ -179,13 +179,14 @@ bool BitBlitAlpha(uint8_t *dst, uint32_t pitchDst,uint8_t *src,uint32_t pitchSrc
  */
 bool BitBlit(uint8_t *dst, uint32_t pitchDst,uint8_t *src,uint32_t pitchSrc,uint32_t width, uint32_t height)
 {
-#if 1
+#if 0
+    // av_image_copy_plane is just a wrapper around a wrapper of line-by-line memcpy
```

Comment:
The comment from https://github.com/mean00/avidemux2/commit/6b5b805826a603935742de5c2c838e2ac05dba63 points to av_image_copy_plane wrapping SSE4 `image_copy_plane_uc_from` from libavutil/x86/imgutils.asm. Are you sure it is not faster than a plain memcpy, given the CPU supports SSE4 and Avidemux doesn't tell FFmpeg not to use it?

Reply:
av_image_copy_plane calls image_copy_plane (without the uc_from extension, which is used for unCached memory)

static void image_copy_plane(uint8_t       *dst, ptrdiff_t dst_linesize,
                             const uint8_t *src, ptrdiff_t src_linesize,
                             ptrdiff_t bytewidth, int height)
{
    if (!dst || !src)
        return;
    av_assert0(FFABS(src_linesize) >= bytewidth);
    av_assert0(FFABS(dst_linesize) >= bytewidth);
    for (;height > 0; height--) {
        memcpy(dst, src, bytewidth);
        dst += dst_linesize;
        src += src_linesize;
    }
}

Reply:
Oops, indeed, now I wonder whether the same reading glitch happened to the author of https://github.com/mean00/avidemux2/commit/6b5b805826a603935742de5c2c838e2ac05dba63 ...

The uc_from implementation should be the one which was hoped to help with slow download speed from VA-API hw surfaces, if I am not mistaken.

Reply:
yes